test/ctia/entity/vulnerability_test.clj

```diff
@@ -47,3 +51,41 @@
                             {:entity-minimal new-vulnerability-minimal
                              :enumerable-fields sut/vulnerability-enumerable-fields
                              :date-fields sut/vulnerability-histogram-fields})))
+
+(deftest test-build-configurations-query
+  (is (= (sut/build-configurations-query (map cpe/->cpe-match ["cpe:2.3:o:juniper:advanced_threat_prevention:*:*:*:*:*:*:*:*"
+                                                               "cpe:2.3:h:juniper:atp400:-:*:*:*:*:*:*:*"]))
+         "configurations.nodes.cpe_match.cpe23Uri:cpe\\:2.3\\:o\\:juniper\\:advanced_threat_prevention* OR configurations.nodes.children.cpe_match.cpe23Uri:cpe\\:2.3\\:o\\:juniper\\:advanced_threat_prevention* OR configurations.nodes.cpe_match.cpe23Uri:cpe\\:2.3\\:h\\:juniper\\:atp400* OR configurations.nodes.children.cpe_match.cpe23Uri:cpe\\:2.3\\:h\\:juniper\\:atp400*")))
+
+(deftest search-by-cpe-match-strings-test
+  (es-helpers/for-each-es-version
+   "List Vulnerabilities with configurations matching CPE 2.3 match strings"
+   [5 7]
+   #(ductile.index/delete! % "*ctia_vulnerability*")
+   (helpers/with-properties ;; simple way to have a proper store initialization
+     ["ctia.store.es.default.port" es-port
+      "ctia.store.es.default.version" version
+      "ctia.store.es.default.auth" es-helpers/basic-auth
+      "ctia.auth.type" "allow-all"]
+     (helpers/fixture-ctia-with-app
+      (fn [app]
+        (let [_ (es-helpers/load-file-bulk app conn "./test/data/indices/sample-vulnerability-150.json")
+              {{:keys [get-store]} :StoreService} (app->APIHandlerServices app)
+              identity-map {:login "johndoe"
+                            :groups ["Administators"]}
+              store (get-store :vulnerability)
+              vulnerability-ids (sut/vulnerability-ids-affected-by-cpe-matches
+                                 {:cpe-matches (map cpe/->cpe-match
+                                                    ["cpe:2.3:h:juniper:ex8200\\/vc_\\(xre\\):-:*:*:*:*:*:*:*"
+                                                     "cpe:2.3:o:juniper:junos:14.1x53:*:*:*:*:*:*:*"])
+                                  :identity-map identity-map
+                                  :query-params {}
+                                  :store store})
+              _ (is (not (realized? vulnerability-ids)))
+              results (sut/search-by-ids {:identity-map identity-map
+                                          :ids vulnerability-ids
+                                          :limit 10
+                                          :params {:sort {:id :asc}}
+                                          :store store})]
+          (is (= #{"CVE-2018-0007" "CVE-2018-0024" "CVE-2018-0031"}
+                 (set (map :title (:data results)))))))))))
```

Comment:
I would benefit of the context of this whole test to add an end to end test fn with calls to the cpe search route. Could you add some representative cases?